test/conformance/memory/urMemBufferMigrateAcrossDevices.cpp

```diff
@@ -0,0 +1,259 @@
+// Copyright (C) 2023 Intel Corporation
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+// See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// Some tests to ensure implicit memory migration of buffers across devices
+// in the same context.
+
+#include "uur/fixtures.h"
+
+using T = uint32_t;
+
+struct urMultiDeviceContextTest : uur::urPlatformTest {
+    void SetUp() {
+        uur::urPlatformTest::SetUp();
+        ASSERT_SUCCESS(urDeviceGet(platform, UR_DEVICE_TYPE_ALL, 0, nullptr,
+                                   &num_devices));
+        ASSERT_NE(num_devices, 0);
+        if (num_devices == 1) {
+            return;
```

Comment:
Isn't there a `GTEST_SKIP()` or something we could use here instead so the test doesn't show passing when it doesn't actually run?

Reply:
Good question. Will check

Reply:
Yep looks like `GTEST_SKIP` works. Have changed.